hawk/app/models/report.rb

```diff
@@ -1,69 +1,225 @@
+                        basename: File.basename(path, ".bz2"),
+                        filename: File.basename(path),
+                        path: path.sub("#{hb_report.path}/", ''),  # only use relative portion
+                        node: path.split(File::SEPARATOR)[-3],
+                        version: version)
+        end
+        # sort is going to be off for identical mtimes (stripped back to the second),
+        # so need secondary sort by filename
+      end
+
+      # add errors to output
+      errors = hb_report.err_filtered
+      errors.each do |err|
+        peinputs.push error: err
+      end
+    end
   end
 
-  def report_path
-    self.class.report_path
+  def peinput(path, basename, node)
+    basename = basename.gsub(/[^\w-]/, "")
+    node = node.gsub(/[^\w_-]/, "")
+    tname = "#{node}/pengine/#{basename}.bz2"
+    path = path.gsub("..", "") # tear out possible relative junk
+    archive.join(path).to_s
   end
 
-  class << self
-    def parse(file)
-      hash = Digest::SHA1.hexdigest(
-        file.basename(".tar.bz2").to_s
-      )
+  def peinput_version(path)
+    nvpair = `CIB_file=#{path} cibadmin -Q --xpath "/cib/configuration//crm_config//nvpair[@name='dc-version']" 2>/dev/null`
+    m = nvpair.match(/value="([^"]+)"/)
+    return nil unless m
+    m[1]
+  end
+
+  def transition_cmd(hb_report, path, cmd)
+    source = archive
+    source = hb_report.path if File.directory?(hb_report.path)
+    Util.capture3("crm", "history", stdin_data: "source #{source}\n#{cmd}\n")
+  end
+
+  def info(hb_report, path)
+    out, err, status = transition_cmd hb_report, path, "transition #{path} nograph"
+    info = out + err
+    info.strip!
+    info = _("No details available") if info.empty?
+    info.insert(0, _("Error:") + "\n") unless status.exitstatus == 0
+    info
+  end
+
+  def tags(hb_report, path)
+    out, err, status = transition_cmd hb_report, path, "transition tags #{path}"
+    out.split
+  end
+
+  def logs(hb_report, path)
+    out, err, status = transition_cmd hb_report, path, "transition log #{path}"
+    info = out + err
+    info.strip!
+    info = _("No details available") if info.empty?
+    info.insert(0, _("Error:") + "\n") unless status.exitstatus == 0
+    info
+  end
+
+  # Returns [success, data|error]
+  def graph(hb_report, path, format=:svg)
+
+    tpath = Pathname.new(hb_report.path).join(path)
+
+    # Apparently we can't rely on the dot file existing in the hb_report, so we
+    # just use ptest to generate it.  Note that this will fail if hacluster doesn't
+    # have read access to the pengine files (although, this should be OK, because
+    # they're created by hacluster by default).
+    require "tempfile"
+    tmpfile = Tempfile.new("hawk_dot")
+    tmpfile.close
+    File.chmod(0666, tmpfile.path)
+    out, err, status = Util.run_as('hacluster', 'crm_simulate', '-x', tpath.to_s, format == :xml ? "-G" : "-D", tmpfile.path.to_s)
```

Comment:
Lint/UselessAssignment: Useless assignment to variable - `out`. Use `_` or `_out` as a variable name to indicate that it won't be used.